packages/expo-notifications/src/getDevicePushTokenAsync.web.ts

```diff
@@ -0,0 +1,111 @@
+import { CodedError, Platform, SyntheticPlatformEmitter } from '@unimodules/core';
+import Constants from 'expo-constants';
+
+import { DevicePushToken } from './getDevicePushTokenAsync';
+
+export default async function getDevicePushTokenAsync(): Promise<DevicePushToken> {
+  const data = await _subscribeUserToPushAsync();
+  SyntheticPlatformEmitter.emit('onDevicePushToken', { devicePushToken: data });
+  return { type: Platform.OS, data };
+}
+
+function guardPermission() {
+  if (!('Notification' in window)) {
+    throw new Error('The Notification API is not available on this device.');
+  }
+  if (!navigator.serviceWorker) {
+    throw new Error(
+      'Notifications cannot be used because the service worker API is not supported on this device. This might also happen because your web page does not support HTTPS.'
+    );
+  }
+  if (Notification.permission !== 'granted') {
+    throw new Error(
+      'Cannot use Notifications without permissions. Please request permissions with `expo-permissions`'
+    );
+  }
+}
+
+async function _subscribeUserToPushAsync(): Promise<DevicePushToken['data']> {
+  if (!Constants.manifest.notification || !Constants.manifest.notification.vapidPublicKey) {
+    throw new CodedError(
+      'E_NOTIFICATIONS_PUSH_WEB_MISSING_CONFIG',
+      'You must provide `notification.vapidPublicKey` in `app.json` to use push notifications on web. Learn more: https://docs.expo.io/versions/latest/guides/using-vapid/.'
+    );
+  }
+  if (!Constants.manifest.notification || !Constants.manifest.notification.serviceWorkerPath) {
+    throw new CodedError(
+      'E_NOTIFICATIONS_PUSH_WEB_MISSING_CONFIG',
+      'You must provide `notification.serviceWorkerPath` in `app.json` to use push notifications on web. Please provide path to the service worker that will handle notifications.'
+    );
+  }
+  guardPermission();
+
+  let registration: ServiceWorkerRegistration | null = null;
+  try {
+    registration = await navigator.serviceWorker.register(
+      Constants.manifest.notification.serviceWorkerPath
+    );
+  } catch (error) {
+    throw new Error(
+      `Notifications might not be working because the service worker (${Constants.manifest.notification.serviceWorkerPath}) couldn't be registered: ${error}`
+    );
+  }
+  await navigator.serviceWorker.ready;
+
+  if (!registration.active) {
+    throw new Error(
```

Comment:
```
  throw new CodedError(
    'ERR_NOTIFICATIONS_PUSH_REGISTRATION_FAILED',
    `Could not register this device for push notifications because the service worker is not active`
  );
```